docs/integrations/positioning-systems/ublox-evk-m8n.mdx

```diff
@@ -0,0 +1,267 @@
+  width="587"
+  height="112"
+/>
+
+Now, our `idVendor` and `idProduct` are the two strings after "ID" divided by
+colon. In our case, `idVendor` is `1546` and `idProduct` is `01a8`.
+
+After getting those two, the only thing that is left is the port that the device
+connects to. You can get it by running in the terminal the command:
+
+```bash
+dmesg -w
+```
+
+and then connecting the module to the rover. You should see that the new device
+is registered, and on what port (with this command, you can also get the
+`idVendor` and `idProduct` values).
+
+<ImageZoom
+  src="/img/robots/leo/integrations/ublox-gps/terminal-ttyacm0.webp"
+  width="867"
+  height="112"
+/>
+
+Now, you need to make a `udev` rules file.
+
+```xml title="/etc/udev/rules.d/ublox.rules"
+KERNEL=="ttyACM*", ATTRS{idVendor}=="1546", ATTRS{idProduct}=="01a8", MODE="0666", GROUP="dialout", SYMLINK+="ublox_gps"
+```
+
+Paste these lines to `/etc/udev/rules.d/ublox.rules` file and reload `udev`
+rules by typing:
+
+```bash
+sudo udevadm control --reload-rules && sudo udevadm trigger
+```
+
+Your device should now be available at the `/dev/ublox_gps` path.
+
+We want the sensor functionality to be available in the ROS ecosystem, so you
+should install a ROS package that provides a node for the sensor you are trying
+to integrate.
+
+```bash
+sudo apt install ros-${ROS_DISTRO}-ublox
+```
+
+Now, create a launch file that would start the node with a fitting
+configuration.
+
+```xml title="/etc/ros/gps.launch.xml"
+<launch>
+  <node pkg="ublox_gps" exec="ublox_gps_node" output="screen">
+    <param from="$(find-pkg-share ublox_gps)/config/c94_m8p_rover.yaml" />
+    <param name="device" value="/dev/ublox_gps"/>
+    <param name="frame_id" value="ublox_frame"/>
+  </node>
+</launch>
+```
+
+In `/etc/ros/robot.launch.xml` add this line somewhere before the closing
+`</launch>` tag, so that your node will start at boot.
+
+```xml title="/etc/ros/robot.launch.xml"
+<include file="/etc/ros/gps.launch.xml"/>
+```
+
+Your robot should be aware of where the module is located and what space it
+occupies. You can ensure it does that by creating a URDF model of the module.
+
+```xml title="/etc/ros/urdf/ublox.urdf"
+<?xml version="1.0"?>
+<robot>
+  <link name="ublox_frame"/>
+
+  <joint name="ublox_joint" type="fixed">
+    <origin xyz="-0.05 0 0"/>
+    <parent link="base_link"/>
+    <child link="ublox_frame"/>
+  </joint>
+
+</robot>
+```
+
+:::warning
+
+Make sure that `frame_id` param in the launch file and link `name` in
+`ublox.urdf` file have the same value.
+
+:::
+
+And including it in the description that is uploaded at boot. Add this line
+somewhere before the closing `</robot>` tag.
+
+```xml title="/etc/ros/urdf/robot.urdf.xacro"
+<xacro:include filename="/etc/ros/urdf/ublox.urdf"/>
+```
+
+The last step is to either reboot the robot
+
+```bash
+reboot
+```
+
+or restart the nodes
+
+```bash
+ros-nodes-restart
+```
```

Comment:
IMO it is not worth recommending reboot.
```suggestion
The last step is to restart the nodes

```bash
ros-nodes-restart
```
```

Reply:
OKay, but it's mentioned probably in every tutorial.